packages/react-instantsearch-core/src/connectors/connectConfigureRelatedItems.ts

```diff
@@ -0,0 +1,220 @@
+import algoliasearchHelper, {
+  PlainSearchParameters,
+  SearchParameters,
+} from 'algoliasearch-helper';
+import createConnector, { ConnectedProps } from '../core/createConnector';
+import {
+  omit,
+  getObjectType,
+  removeEmptyKey,
+  removeEmptyArraysFromObject,
+} from '../core/utils';
+import {
+  refineValue,
+  getIndexId,
+  hasMultipleIndices,
+} from '../core/indexUtils';
+
+type Hit = any;
+
+export type MatchingPatterns = {
+  [attribute: string]: {
+    /**
+     * The score of the optional filter.
+     *
+     * @see https://www.algolia.com/doc/guides/managing-results/rules/merchandising-and-promoting/in-depth/optional-filters/
+     */
+    score: number;
+  };
+};
+
+interface ConfigureRelatedItemsProps {
+  /**
+   * The reference hit to extract the filters from.
+   */
+  hit: Hit;
```

Comment:
It's somewhat late now to make this change, but what if we allow `hits: Hit[]`, so people can create related hits when a product page or home page should be related to multiple items?

We can still consider this later before we move away from `EXPERIMENTAL_`

Reply:
Do you mean to create `optionalFilters` based on multiple hits?
For example, if there're two hits(Apple, Amazon), then `optionalFilters` becomes like this?
```
[
    ["brand:Apple<score=3> OR brand:Amazon<score=3>"]
]
```

Reply:
Interesting! Definitely worth exploring before getting out of the experimental phase.